### PR TITLE
Add extractMetricName method

### PR DESCRIPTION
**Pull request by MSDehghan**

### CODE REVIEW

File: src/main/java/ir/sahab/dropwizardmetrics/LabeledMetric.java [hunk 1]
```diff
@@ -54,6 +54,16 @@ public static boolean hasLabel(String nameAndLabels) {
...
+    /**
+     * Extracts the name from a labeled metric name.
+     */
+    public static String extractMetricName(String labeledMetricName) {
+        if (!hasLabel(labeledMetricName)) {
+            return labeledMetricName;
+        }
+        return labeledMetricName.substring(0, labeledMetricName.indexOf("["));
+    }
+
```
Comment (bozorgzadeh) [inline src/main/java/ir/sahab/dropwizardmetrics/LabeledMetric.java:66]: Use 'labeledMetricName' consistently for the argument of other methods too:

hasLabel(String labeledMetricName)
private constructor: LabeledMetric(String labeledMetricName)
processLabels(String labeledMetricName, Consumer...)

File: src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java [hunk 2]
```diff
@@ -123,7 +113,7 @@ private String extractMetricName(String labeledMetricName) {
      * write "metric_name[type=Thread, name=DGC] (with a space after the comma) because it will be interpreted as having
      * a key called " name", with a leading space in the name.
      */
-    private String extractQuotedLabelValues(String metricName) {
+    private static String extractQuotedLabelValues(String metricName) {
```
Comment (bozorgzadeh) [inline src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java]:
```suggestion
    private static String extractQuotedLabelValues(String labeledMetricName) {
```

File: src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java [hunk 3]
```diff
@@ -105,16 +105,6 @@ private String quoteValueIfRequired(String propertyValue) {
...
-     */
-    private String extractMetricName(String labeledMetricName) {
-        if (!LabeledMetric.hasLabel(labeledMetricName)) {
-            return labeledMetricName;
-        }
-        return labeledMetricName.substring(0, labeledMetricName.indexOf("["));
-    }
-
     /**
      * Extract and quotes values of label if required. If metricName is not a labeled metric, the returned value
```
Comment (bozorgzadeh) [inline src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java]:
```suggestion
     * Extracts and quotes values of label if required. If metric has labels, the returned value
```

File: src/main/java/ir/sahab/dropwizardmetrics/LabeledMetric.java (hunk 1 above)
Comment (MSDehghan) [reply src/main/java/ir/sahab/dropwizardmetrics/LabeledMetric.java:66]: Done

File: src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java (hunk 2 above)
Comment (MSDehghan) [reply src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java]: Done

File: src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java (hunk 3 above)
Comment (MSDehghan) [reply src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java]: Done

**Comment by MSDehghan** [comment]
Done

**Pull request merged**